product.py

```diff
@@ -109,6 +119,26 @@ class Product(WineMixin, metaclass=PoolMeta):
                     'bottled-wine']
             )}, depends=['agronomic_type']), 'get_alcohol_volume')
 
+    @classmethod
+    def deactivate_no_stock_variants_cron(cls):
+        pool = Pool()
+        Location = pool.get('stock.location')
+        ProductConfiguration = pool.get('product.configuration')
+        config = ProductConfiguration(1)
+        locations = Location.search(['type', '=', 'warehouse'])
+        locations = [location.id for location in locations]
+        with Transaction().set_context(locations=locations, with_childs=True):
+            products = cls.search(
+                [
+                    ('quantity', '=', 0),
+                    ('template.variant_deactivate_stock_zero', '=', True),
+                    ('create_date', '<',
+                        (datetime.now() - config.variant_deactivation_time))
```

Comment:
You should check if variant_deactivation_time is None and do nothing instead of crashing.